File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineRetakeExam2And3May2019/03.MobileOperator.js
function mobileOperation(input) {
    let periodOfContract = input[0];
    let typeOfContract = input[1];
    let mobileInternet = input[2];
    let numbersOfMonth = Number(input[3]);

    let pricePerMonth = 0;

    switch(periodOfContract) {
        case 'one':
            switch(typeOfContract) {
                case 'Small': pricePerMonth = 9.98; break;
                case 'Middle': pricePerMonth = 18.99; break;
                case 'Large': pricePerMonth = 25.98; break;
                case 'ExtraLarge': pricePerMonth = 35.99; break;
            } break;

        case 'two':
            switch(typeOfContract) {
                case 'Small': pricePerMonth = 8.58; break;
                case 'Middle': pricePerMonth = 17.09; break;
                case 'Large': pricePerMonth = 23.59; break;
                case 'ExtraLarge': pricePerMonth = 31.79; break;
            } break;
    }

    if(mobileInternet === 'yes') {
        if(pricePerMonth <= 10) {
            pricePerMonth += 5.50;
        } else if (pricePerMonth <= 30) {
            pricePerMonth += 4.35;
        } else if (pricePerMonth > 30) {
            pricePerMonth += 3.85;
        }
    }

    let totalCost = pricePerMonth * numbersOfMonth;

    if(periodOfContract === 'two') {
        totalCost = totalCost * 0.9625;
    }

    console.log(`${totalCost.toFixed(2)} lv.`);
}

mobileOperation(["one",
"Small",
"yes",
"10"])

mobileOperation(["two",
"Large",
"no",
"10"])

File: Programming_Basics_with_Python/programming_basics_online_retake_exam_2_and_3_may_2019/03._mobile_operator.py
contract_term = input()
type_of_contract = input()
added_mobile_internet = input()
number_of_months_to_pay = int(input())

price_per_month = 0
if contract_term == 'one':
    if type_of_contract == 'Small':
        price_per_month = 9.98
    elif type_of_contract == 'Middle':
        price_per_month = 18.99
    elif type_of_contract == 'Large':
        price_per_month = 25.98
    elif type_of_contract == 'ExtraLarge':
        price_per_month = 35.99
elif contract_term == 'two':
    if type_of_contract == 'Small':
        price_per_month = 8.58
    elif type_of_contract == 'Middle':
        price_per_month = 17.09
    elif type_of_contract == 'Large':
        price_per_month = 23.59
    elif type_of_contract == 'ExtraLarge':
        price_per_month = 31.79

if added_mobile_internet == 'yes':
    if price_per_month <= 10:
        price_per_month += 5.50
    elif price_per_month <= 30:
        price_per_month += 4.35
    else:
        price_per_month += 3.85

money_for_pay = number_of_months_to_pay * price_per_month
if contract_term == 'two':
    money_for_pay = money_for_pay * 0.9625
print(f'{money_for_pay:.2f} lv.')